src/tmpfiles/tmpfiles.c

```diff
@@ -3743,6 +3754,10 @@ static int parse_argv(int argc, char *argv[]) {
                         arg_boot = true;
                         break;
 
+                case ARG_FACTORY_RESET:
+                        arg_factory_reset = true;
+                        break;
+
```

Comment:
I can change the option to take a boolean - the rest sounds like it can be done separately? I want to use this to let packages define things that need to be done on 'purge' (which is different from 'remove')

Reply:
I assume you are referring to the distinction between remove and purge in Debian.
While I like the idea to have a mechanism to cleanup resources when a package is uninstalled in a declarative way, I'm unsure how this could be done via tmpfiles. Those files are removed during the "uninstall" phase, so they might no longer be available during "purge".

Reply:
That's our problem to figure out - we can do it in the prerm for example

Reply:
We could copy it out to /var/ somewhere in the prerm, and call it in the postrm+purge, or something like that. We don't have to solve it here - the important thing here is to have the functionality available

Reply:
> I can change the option to take a boolean - the rest sounds like it can be done separately? I want to use this to let packages define things that need to be done on 'purge' (which is different from 'remove')

I don't understand what this means? I sense this is an entirely different usecase for factory reset than the systemd-repart thing?

can you elaborate the usecase, I really don't grok this?

Reply:
i don't get it? why involve tmpfiles in package management? doesn't dpkg track file ownership?

Reply:
it's for things that are _created_ with tmpfiles.d - ie, you ship a tmpfiles to create stuff in /var/, and you add a `R@` line so that it gets removed on purge, but not when the package is simply removed, or `tmpfiles --remove` is ran for other reasons

Reply:
but why doesn't dpkg track that? on rpm you gave "ghost" entries in RPM ownership definition. dpkg doesn't have that?

Reply:
no, there's no such thing

Reply:
that sounds like something to fix in dpkg though, no?

Reply:
yeah that's not, er, likely to happen